lib/ansible/plugins/loader.py

```diff
@@ -867,8 +874,15 @@ def get_with_context(self, name, *args, **kwargs):
             # FIXME: this is probably an error (eg removed plugin)
             return get_with_context_result(None, plugin_load_context)
 
+        return self.get_from_context(plugin_load_context, *args, **kwargs)
+
+    def get_from_context(self, plugin_load_context, *args, **kwargs):
```

Comment:
I see the need to split some of the plugin construction from plugin location for specific cases where we're serving it from a cache, but pretty sure we don't want to turn all of this into public API- it's *really* special-purpose. I'm wondering if it makes more sense to just special case short-circuit instance-cached plugins internally early on in `get_with_context`, and special-case a tiny bit of `all` to internally remember it so we're not having to leak the specifics of yet another PluginLoader cache behavior out to the call sites...

Reply:
Fixed.